src/golang/lib/workflow/dag/workflow_dag.go

```diff
@@ -30,6 +30,9 @@ type WorkflowDag interface {
 	// OperatorInputs returns all the artifacts that are fed as input to the given operator.
 	OperatorInputs(operator.Operator) ([]artifact.Artifact, error)
 
+	// Operators that generate the input artifacts of a given operator.
+	OperatorParents(operator.Operator) ([]operator.Operator, error)
```

Comment:
@saurav-c can you review the changes to this file in particular? Wanted to make sure that the additions follow the patterns. I originally was going to do something like ArtifactParent and push the finding of the parent logic to the DatabricksJobManager, but thought any logic involving the operators on a `workflowdag` belong here. Let me know what you think.

Reply:
I think this makes sense to implement here